### ISSUE TITLE
Switch from the custom unit test harness to TSLint's --test

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [microsoft/tslint-microsoft-contrib#489]
It'd be nice to not have to maintain the custom unit test framework, and instead use the same one as TSLint core's.

**Comment by IllusionMH** [comment]
I'd like to take on this.

**Comment by IllusionMH** [comment]
Update on this issue.

After #584 and #616 landed, now all infrastructure tasks are ready to accomplish test migration.

This repo has 92 rules and it will definitely take time to properly migrate all tests. I will do it in several PRs.

**If you want to help - you are welcome**. Plase leave a comment with list of rules you want to migrate, so we won't do same work.

Progress: **1 of92**

🏃  marks rules that are in batch that I'm working on.

- [ ] `chai-prefer-contains-to-index-of`
- [ ] `chai-vague-errors`
- [ ] :running_man: `export-name`
- [ ] `function-name`
- [ ] `import-name`
- [ ] `informative-docs`
- [ ] `insecure-random`
- [ ] `jquery-deferred-must-complete`
- [ ] `max-func-body-length`
- [ ] `missing-jsdoc`
- [ ] `missing-optional-annotation`
- [ ] `mocha-avoid-only`
- [ ] `mocha-no-side-effect-code`
- [ ] `mocha-unneeded-done`
- [ ] `no-backbone-get-set-outside-model`
- [ ] `no-banned-terms`
- [ ] `no-constant-condition`
- [ ] `no-control-regex`
- [ ] :running_man: `no-cookies`
- [ ] `no-delete-expression`
- [ ] `no-disable-auto-sanitization`
- [ ] `no-document-domain`
- [ ] `no-document-write`
- [ ] `no-duplicate-case`
- [ ] `no-duplicate-parameter-names`
- [ ] `no-empty-interfaces`
- [ ] `no-empty-line-after-opening-brace`
- [ ] `no-exec-script`
- [ ] `no-for-in`
- [ ] :running_man: `no-function-constructor-with-string-args`
- [ ] :running_man: `no-function-expression`
- [ ] `no-http-string`
- [ ] `no-increment-decrement`
- [ ] `no-inner-html`
- [ ] `no-invalid-regexp`
- [ ] `no-jquery-raw-elements`
- [ ] `no-missing-visibility-modifiers`
- [ ] `no-multiline-string`
- [ ] `no-multiple-var-decl`
- [ ] :running_man: `no-octal-literal`
- [ ] `no-regex-spaces`
- [X] `no-relative-imports`
- [ ] :running_man: `no-reserved-keywords`
- [ ] `no-single-line-block-comment`
- [ ] `no-stateless-class`
- [ ] :running_man: `no-string-based-set-immediate`
- [ ] :running_man: `no-string-based-set-interval`
- [ ] :running_man: `no-string-based-set-timeout`
- [ ] `no-suspicious-comment`
- [ ] `no-typeof-undefined`
- [ ] `no-unexternalized-strings`
- [ ] `no-unnecessary-bind`
- [ ] `no-unnecessary-field-initialization`
- [ ] `no-unnecessary-local-variable`
- [ ] `no-unnecessary-override`
- [ ] :running_man: `no-unnecessary-semicolons`
- [ ] `no-unsupported-browser-code`
- [ ] `no-useless-files`
- [ ] `no-var-self`
- [ ] `no-with-statement`
- [ ] `non-literal-fs-path`
- [ ] `non-literal-require`
- [ ] `possible-timing-attack`
- [ ] `prefer-array-literal`
- [ ] :running_man: `prefer-type-cast`
- [ ] `promise-must-complete`
- [ ] `react-a11y-anchors`
- [ ] `react-a11y-aria-unsupported-elements`
- [ ] `react-a11y-event-has-role`
- [ ] `react-a11y-image-button-has-alt`
- [ ] :running_man: `react-a11y-img-has-alt`
- [ ] `react-a11y-input-elements`
- [ ] `react-a11y-lang`
- [ ] `react-a11y-meta`
- [ ] `react-a11y-no-onchange`
- [ ] :running_man: `react-a11y-props`
- [ ] :running_man: `react-a11y-proptypes`
- [ ] `react-a11y-required`
- [ ] :running_man: `react-a11y-role`
- [ ] :running_man: `react-a11y-role-has-required-aria-props`
- [ ] :running_man: `react-a11y-role-supports-aria-props`
- [ ] :running_man: `react-a11y-tabindex-no-positive`
- [ ] `react-a11y-titles`
- [ ] `react-anchor-blank-noopener`
- [ ] `react-iframe-missing-sandbox`
- [ ] `react-no-dangerous-html`
- [ ] :running_man: `react-this-binding-issue`
- [ ] `react-tsx-curly-spacing`
- [ ] `react-unused-props-and-state`
- [ ] `underscore-consistent-invocation`
- [ ] `use-named-parameter`
- [ ] `valid-typeof`

**Comment by JoshuaKGoldberg** [comment]
I'll roughly start on the rules from the bottom up today, ending somewhere before `no-unnecessary-bind`, then post the list here.

**Comment by JoshuaKGoldberg** [comment]
Changing the label from `Difficulty: Hard` to `Difficulty: Easy` now that the infrastructure _(and endline behaviors)_ have been figured out. 🚀

**Comment by JoshuaKGoldberg** [comment]
Ha oops...

**Comment by JoshuaKGoldberg** [comment]
☠️  *It's time!* ☠️ 

Per #876, this repository is no longer accepting feature pull requests. TSLint is being deprecated and we recommend you switch to https://typescript-eslint.io.

Thanks for open sourcing with us, everyone!
